pkg/services/extsvcauth/extsvcaccounts/service.go

```diff
@@ -41,6 +52,41 @@ func (esa *ExtSvcAccountsService) RetrieveExtSvcAccount(ctx context.Context, org
 	}, nil
 }
 
+// SaveExternalService creates, updates or delete a service account (and its token) with the requested permissions.
+func (esa *ExtSvcAccountsService) SaveExternalService(ctx context.Context, cmd *extsvcauth.ExternalServiceRegistration) (*extsvcauth.ExternalService, error) {
```

Comment:
This is to implement the `extsvcauth.ExternalServiceRegistry` interface.
We will use this in another PR to allow plugins to request service account tokens.